pkg/driver/docker/docker.go

```diff
@@ -363,6 +366,188 @@ func (d *dockerDriver) RunDockerDevContainer(
 		return err
 	}
 
+	if runtime.GOOS != "windows" && ((parsedConfig.ContainerUser != "" || parsedConfig.RemoteUser != "") &&
+		(parsedConfig.UpdateRemoteUserUID == nil || *parsedConfig.UpdateRemoteUserUID)) {
+		// Retrieve local user UID and GID
+		localUser, err := user.Current()
+		if err != nil {
+			return err
+		}
+		localUid := localUser.Uid
+		localGid := localUser.Gid
+
+		// Retrieve user to update
+		containerUser := parsedConfig.RemoteUser
+		if containerUser == "" {
+			containerUser = parsedConfig.ContainerUser
+		}
+		container, err := d.FindDevContainer(ctx, workspaceId)
+		if err != nil {
+			return err
+		} else if container == nil {
+			return nil
+		}
+
+		// Create temporary files to store /etc/passwd and /etc/group
+		containerPasswdFileIn, err := os.CreateTemp("", "devpod_container_passwd_in")
+		if err != nil {
+			return err
+		}
+		containerGroupFileIn, err := os.CreateTemp("", "devpod_container_group_in")
+		if err != nil {
+			return err
+		}
+		containerPasswdFileOut, err := os.CreateTemp("", "devpod_container_passwd_out")
+		if err != nil {
+			return err
+		}
+		containerGroupFileOut, err := os.CreateTemp("", "devpod_container_group_out")
+		if err != nil {
+			return err
+		}
+		defer os.Remove(containerPasswdFileIn.Name())
```

Comment:
each defer needs to proceed the file creation, otherwise the last file could fail and we would have the first 4 not cleaned up

Reply:
Agreed. Changed it.